connection_configuration.go

```diff
@@ -0,0 +1,475 @@
+				return err
+			}
+		case "disableconsolelogin":
+			if vv, parsingErr = parseBool(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+			if vv {
+				cfg.DisableConsoleLogin = ConfigBoolTrue
+			} else {
+				cfg.DisableConsoleLogin = ConfigBoolFalse
+			}
+		case "disablesamlurlcheck":
+			if vv, parsingErr = parseBool(value); parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+			if vv {
+				cfg.DisableSamlURLCheck = ConfigBoolTrue
+			} else {
+				cfg.DisableSamlURLCheck = ConfigBoolFalse
+			}
+		case "token_file_path":
+			tokenPath, parsingErr = parseString(value)
+			if parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+		default:
+			var param string
+			param, parsingErr = parseString(value)
+			if parsingErr != nil {
+				err.MessageArgs = []interface{}{key, value}
+				return err
+			}
+			cfg.Params[urlDecodeIfNeeded(key)] = &param
+		}
+	}
+	if shouldReadTokenFromFile(cfg) {
+		v, err := readToken(tokenPath)
+		if err != nil {
+			return err
+		}
+		cfg.Token = v
+	}
+	return nil
+}
+
+func parseInt(i interface{}) (int, error) {
+	if _, ok := i.(string); !ok {
+		if _, ok := i.(int); !ok {
+			return 0, errors.New("parse Error")
+		}
+		num := i.(int)
+		return num, nil
+	}
+	v := i.(string)
+	num, err := strconv.Atoi(v)
+
+	if err != nil {
+		return num, err
+	}
+	return num, nil
+}
+
+func parseBool(i interface{}) (bool, error) {
+	v, ok := i.(string)
+	if !ok {
+		if _, ok := i.(bool); !ok {
+			return false, errors.New("parse Error")
+		}
+		vv := i.(bool)
+		return vv, nil
+	}
+	vv, err := strconv.ParseBool(v)
+	if err != nil {
+		return false, errors.New("parse Error")
+	}
+	return vv, nil
+}
+
+func parseDuration(i interface{}) (time.Duration, error) {
+	v, ok := i.(string)
+	if !ok {
+		num, err := parseInt(i)
+		if err != nil {
+			return time.Duration(0), err
+		}
+		t := int64(num)
+		return time.Duration(t * int64(time.Second)), nil
+	}
+	t, err := strconv.ParseInt(v, 10, 64)
+	if err != nil {
+		return time.Duration(0), err
+	}
+	return time.Duration(t * int64(time.Second)), nil
+}
+
+func readToken(tokenPath string) (string, error) {
+	if tokenPath == "" {
+		tokenPath = "./snowflake/session/token"
+	}
+	if !path.IsAbs(tokenPath) {
+		snowflakeConfigDir, err := getTomlFilePath(os.Getenv("SNOWFLAKE_HOME"))
+		if err != nil {
+			return "", err
+		}
+		tokenPath = path.Join(snowflakeConfigDir, tokenPath)
+	}
+	err := validateFilePermission(tokenPath)
+	if err != nil {
+		return "", err
+	}
+	token, err := os.ReadFile(tokenPath)
+	if err != nil {
+		return "", err
+	}
+	return string(token), nil
+}
+
+func parseString(i interface{}) (string, error) {
+	v, ok := i.(string)
+	if !ok {
+		return "", errors.New("Error")
+	}
+	return v, nil
+}
+
+func getTomlFilePath(filePath string) (string, error) {
+	var dir string
+	if len(filePath) != 0 {
+		dir = filePath
```

Comment:
Why store filePath in another variable? filePath is not used below.

Reply:
I removed dir and use filePath.